packages/cdktf-cli/src/bin/cmds/handlers.ts

```diff
@@ -417,21 +420,29 @@ export async function providerAdd(argv: any) {
     config.projectDirectory
   );
 
-  let needsGet = false;
-
-  for (const provider of argv.provider) {
+  const providerAddPromises = argv.provider.map(async (provider: string) => {
     const constraint = ProviderConstraint.fromConfigEntry(provider);
 
     if (argv.forceLocal) {
-      needsGet = true;
       await manager.addLocalProvider(constraint);
+      return true;
     } else {
       const { addedLocalProvider } = await manager.addProvider(constraint);
-      if (addedLocalProvider) {
-        needsGet = true;
-      }
+      return addedLocalProvider;
```

Comment:
As discussed offline, we should confirm that we don't run into race conditions like e.g. running `npm install` twice at the same time.